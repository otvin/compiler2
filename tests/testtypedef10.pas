program testtypedef10(output);
{testing subranges when one of the identifiers is overridden in a proc}

type
	Color = (Red, Green, Blue, Orange);
	AnotherColor = Color;
	myrange = Red..Blue;

procedure printcolor(c:anothercolor);
begin
	if c = Red then
		writeln ('Red');
	if c = Green then
		writeln ('Green');
	if c = Blue then
		writeln ('Blue');
	if c = Orange then
		writeln ('Orange');
end;

procedure foo(i:integer);
var Blue:integer; q:myrange;
begin
	Blue := i;
	writeln(Blue);
	q := Red;
	printcolor(q);
	q := succ(q);
	printcolor(q);
	q := succ(q);
	printcolor(q);
	writeln('this should error');
	q := succ(q);
	printcolor(q);
end;

begin
	foo(3);
end.
