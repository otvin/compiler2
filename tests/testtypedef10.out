3
Red
Green
Blue
this should error
Error: value exceeds range for subrange type